packages/hydrogen/src/csp/csp.ts

```diff
@@ -133,10 +133,12 @@ function createCSPHeader(
   for (const key in defaultDirectives) {
     const directive = directives[key as keyof CreateContentSecurityPolicy];
     if (key && directive) {
-      combinedDirectives[key] = addCspDirective(
-        directive,
-        defaultDirectives[key],
-      );
+      combinedDirectives[key] =
+        // Since `frame-ancestors` defaults to `none`, don't 
+        // merge the override with the default value.
+        key === 'frameAncestors'
+          ? directive
+          : addCspDirective(directive, defaultDirectives[key]);
```

Comment:
Just to future proof a little bit (in case default ever change)
 Maybe something like this?

`key === 'frameAncestors' && defaultDirectives[key].includes("'none'")` 

Also this way you likely wont need to add the comment. The code will be self documenting.

Reply:
Great suggestion! I'll add this and adjust the test case as well